src/async/async_api.c

```diff
@@ -540,8 +539,13 @@ __wt_async_flush(WT_SESSION_IMPL *session)
 	WT_ASSERT(session, async->flush_op.state == WT_ASYNCOP_FREE);
 	async->flush_op.state = WT_ASYNCOP_READY;
 	WT_RET(__wt_async_op_enqueue(session, &async->flush_op));
-	while (async->flush_state != WT_ASYNC_FLUSH_COMPLETE)
-		__wt_cond_wait(session, async->flush_cond, 100000);
+	while (async->flush_state != WT_ASYNC_FLUSH_COMPLETE) {
+		/*
+		 * No quit function needed, we're only pausing for 1/10th of a
+		 * second.
+		 */
```

Comment:
> Let's be consistent and call it a "run function" everywhere.

Done.

> Also, consider adding a function __wt_cond_wait_nocheck or similar that asserts the timeout is less than ~1s to avoid repeating this comment everywhere.

I added an assert to ensure we don't sleep for more than 5 seconds w/o a run check function.

I'm hesitant to add a check function. There are 4 flavors of this call (`__wt_cond_wait_signal()` and its wrappers `__wt_cond_wait()`, `__wt_cond_auto_wait_signal()` and `__wt_cond_auto_wait()`), we'd need 4 more names to do that. Let me know if you want that change.